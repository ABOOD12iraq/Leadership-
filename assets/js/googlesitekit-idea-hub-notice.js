/**
 * Idea Hub Notice (for Block Editor) entrypoint.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __, sprintf } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { getItem, setItem } from './googlesitekit/api/cache';

const shownNotices = [];

const editorNoticeKey = ( postID ) => {
	return `modules::idea-hub::dismissed-editor-notice-${ postID }`;
};

const loadIdeaHubNotices = async ( _global = global ) => {
	const { wp } = _global;

	const hasNotice = ( postID ) => {
		if ( wp.data.select( 'core/notices' ).getNotices() === undefined ) {
			return undefined;
		}

		return wp.data.select( 'core/notices' ).getNotices().some( ( notice ) => {
			return notice.id === editorNoticeKey( postID );
		} );
	};

	const listener = async () => {
		// eslint-disable-next-line sitekit/acronym-case
		const postID = wp.data.select( 'core/editor' ).getCurrentPostId();

		if ( ! postID ) {
			return;
		}

		const { cacheHit } = await getItem( editorNoticeKey( postID ) );

		if ( hasNotice( postID ) === false ) {
			// We've already shown this notice, so when it's hidden, mark it as shown
			// so it doesn't appear again.
			if ( shownNotices.includes( editorNoticeKey( postID ) ) ) {
				setItem(
					editorNoticeKey( postID ),
					postMeta.googlesitekitpersistent_idea_text,
					{ ttl: 108000 } // Don't show this notice for another 90 days.
				);
				unsubscribeFromListener();
				return;
			}

			// We've already shown this notice on a previous visit to this page in
			// the editor, so don't show it again.
			if ( cacheHit ) {
				unsubscribeFromListener();
				return;
			}

			// We haven't shown any notice for this post before, so let's check for
			// Idea Hub postmeta.
			const postMeta = wp.data.select( 'core/editor' ).getEditedPostAttribute( 'meta' );

			// eslint-disable-next-line camelcase
			if ( postMeta?.googlesitekitpersistent_idea_text ) {
				wp.data.dispatch( 'core/notices' ).createInfoNotice(
					/* translators: %s: Idea post name */
					sprintf( __( 'This post was created from an idea you picked in Site Kitâ€™s Idea Hub: %s', 'google-site-kit' ), postMeta.googlesitekitpersistent_idea_text ),
					{ id: editorNoticeKey( postID ) }
				);

				// Mark the notice as shown locally, so if it is no longer marked as
				// visible, we known it was dismissed (rather than having yet to be
				// displayed).
				shownNotices.push( editorNoticeKey( postID ) );
			}
		}
	};

	const unsubscribeFromListener = wp.data.subscribe( listener );
};

loadIdeaHubNotices();
