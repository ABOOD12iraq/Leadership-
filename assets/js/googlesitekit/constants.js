/**
 * Core constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const SCREEN_CONTEXT_ACTIVATION = 'activation';
export const SCREEN_CONTEXT_ADMIN_BAR = 'adminBar';
export const SCREEN_CONTEXT_DASHBOARD = 'dashboard';
export const SCREEN_CONTEXT_PAGE_DASHBOARD = 'pageDashboard';
export const SCREEN_CONTEXT_MODULE_DASHBOARD = 'moduleDashboard';
export const SCREEN_CONTEXT_SETTINGS = 'settings';
export const SCREEN_CONTEXT_SPLASH = 'splash';
export const SCREEN_CONTEXT_USER_INPUT = 'userInput';
export const SCREEN_CONTEXT_WP_DASHBOARD = 'wpDashboard';
