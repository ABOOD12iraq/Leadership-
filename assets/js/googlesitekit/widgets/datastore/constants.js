/**
 * `core/widgets` data store: constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const WIDGET_AREA_STYLES = { BOXES: 'boxes', COMPOSITE: 'composite' };
export const WIDGET_WIDTHS = { QUARTER: 'quarter', HALF: 'half', FULL: 'full' };

export const CORE_WIDGETS = 'core/widgets';

// Key Metrics Widgets
export const KM_ANALYTICS_ENGAGED_TRAFFIC_SOURCE =
	'kmAnalyticsEngagedTrafficSource';
export const KM_ANALYTICS_LOYAL_VISITORS = 'kmAnalyticsLoyalVisitors';
export const KM_ANALYTICS_NEW_VISITORS = 'kmAnalyticsNewVisitors';
export const KM_ANALYTICS_POPULAR_CONTENT = 'kmAnalyticsPopularContent';
export const KM_ANALYTICS_POPULAR_PRODUCTS = 'kmAnalyticsPopularProducts';
export const KM_ANALYTICS_TOP_CITIES = 'kmAnalyticsTopCities';
export const KM_ANALYTICS_TOP_CONVERTING_TRAFFIC_SOURCE =
	'kmTopConvertingTrafficSource';
export const KM_ANALYTICS_TOP_COUNTRIES = 'kmAnalyticsTopCountries';
export const KM_ANALYTICS_TOP_TRAFFIC_SOURCE = 'kmAnalyticsTopTrafficSource';
export const KM_SEARCH_CONSOLE_POPULAR_KEYWORDS =
	'kmSearchConsolePopularKeywords';
