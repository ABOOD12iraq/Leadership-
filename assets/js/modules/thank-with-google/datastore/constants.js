/**
 * `modules/thank-with-google` data store constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const MODULES_THANK_WITH_GOOGLE = 'modules/thank-with-google';

export const STATE_ACTIVE = 'ACTIVE';
export const STATE_ACTION_REQUIRED = 'ACTION_REQUIRED';
export const STATE_PENDING_VERIFICATION = 'PENDING_VERIFICATION';

export const BUTTON_PLACEMENT_DYNAMIC_LOW = 'dynamic_low';
export const BUTTON_PLACEMENT_DYNAMIC_HIGH = 'dynamic_high';
export const BUTTON_PLACEMENT_STATIC_AUTO = 'static_auto';
export const BUTTON_PLACEMENT_STATIC_ABOVE_CONTENT = 'static_above-content';
export const BUTTON_PLACEMENT_STATIC_BELOW_CONTENT = 'static_below-content';
export const BUTTON_PLACEMENT_STATIC_BELOW_1ST_PARAGRAPH =
	'static_below-first-paragraph';
