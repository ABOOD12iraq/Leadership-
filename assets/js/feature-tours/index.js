/**
 * Feature Tours.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { VIEW_CONTEXT_DASHBOARD } from '../googlesitekit/constants';
import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';

const allTrafficWidget = {
	slug: 'allTrafficWidget',
	contexts: [ VIEW_CONTEXT_DASHBOARD ],
	version: '1.25.0',
	checkRequirements: async ( registry ) => {
		// Here we need to wait for the underlying selector to be resolved before selecting `isModuleConnected`.
		await registry.__experimentalResolveSelect( CORE_MODULES ).getModules();
		if ( ! registry.select( CORE_MODULES ).isModuleConnected( 'analytics' ) ) {
			return false;
		}
		return true;
	},
	steps: [
		{
			target: '.googlesitekit-widget--analyticsAllTraffic__user-count-chart',
			title: __( "It's now easier to see your site's traffic at a glance", 'google-site-kit' ),
			content: __( 'Check the trend graph to see how your traffic changed over time', 'google-site-kit' ),
		},
		{
			target: '.googlesitekit-widget--analyticsAllTraffic__dimensions-chart',
			title: __( 'See where your visitors come from', 'google-site-kit' ),
			content: __( 'Click on the chart slices to see how each segment has changed over time', 'google-site-kit' ),
		},
		{
			target: '.googlesitekit-header__date-range-selector-menu',
			title: __( 'Check how your traffic changed since you last looked', 'google-site-kit' ),
			content: __( 'Select a time frame to see the comparison with the previous time period', 'google-site-kit' ),
		},
	],
};

// Ordered tours.
export default [
	allTrafficWidget,
];
