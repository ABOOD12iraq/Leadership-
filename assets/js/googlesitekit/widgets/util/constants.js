/**
 * Widgets layout constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { WIDGET_WIDTHS } from '../datastore/constants';

export const WIDTH_GRID_COUNTER_MAP = {
	[ WIDGET_WIDTHS.QUARTER ]: 3,
	[ WIDGET_WIDTHS.HALF ]: 6,
	[ WIDGET_WIDTHS.FULL ]: 12,
};

export const WIDTH_GRID_CLASS_MAP = {
	[ WIDGET_WIDTHS.QUARTER ]: [
		'mdc-layout-grid__cell',
		'mdc-layout-grid__cell--span-2-phone',
		'mdc-layout-grid__cell--span-3-desktop',
		'mdc-layout-grid__cell--span-4-tablet',
	],
	[ WIDGET_WIDTHS.HALF ]: [
		'mdc-layout-grid__cell',
		'mdc-layout-grid__cell--span-6-desktop',
		'mdc-layout-grid__cell--span-8-tablet',
	],
	[ WIDGET_WIDTHS.FULL ]: [
		'mdc-layout-grid__cell',
		'mdc-layout-grid__cell--span-12',
	],
};

export const HIDDEN_CLASS = 'googlesitekit-hidden';
