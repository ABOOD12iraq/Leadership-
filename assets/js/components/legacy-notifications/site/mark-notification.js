/**
 * Internal dependencies
 */
import data, { TYPE_CORE } from '../../data';
import { trackEvent } from '../../../util/tracking';

const ACCEPTED = 'accepted';
const DISMISSED = 'dismissed';

/**
 * Marks the given notification with the provided state.
 *
 * @since 1.4.0
 *
 * @param {string} id    Notification ID.
 * @param {string} state Notification state.
 * @return {Promise} Promise from setting the notification.
 */
export async function markNotification( id, state ) {
	// Invalidate the cache so that notifications will be fetched fresh
	// to not show a marked notification again.
	data.invalidateCacheGroup( TYPE_CORE, 'site', 'notifications' );

	await trackEvent( 'site_notifications', state, id );

	return await data.set( TYPE_CORE, 'site', 'mark-notification', {
		notificationID: id,
		notificationState: state,
	} );
}

/**
 * Marks the given notification as accepted.
 *
 * @since 1.4.0
 *
 * @param {string} id Notification ID.
 * @return {Promise} Promise that is fulfilled after the notification is marked as accepted.
 */
export async function acceptNotification( id ) {
	return await markNotification( id, ACCEPTED );
}

/**
 * Marks the given notification as dismissed.
 *
 * @since 1.4.0
 *
 * @param {string} id Notification ID.
 * @return {Promise} Promise that is fulfilled after the notification is marked as dismissed.
 */
export async function dismissNotification( id ) {
	return await markNotification( id, DISMISSED );
}
