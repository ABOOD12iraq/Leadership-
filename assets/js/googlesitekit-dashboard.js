/**
 * Dashboard component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import { clearWebStorage } from './util';
import Root from './components/Root';
import './components/legacy-notifications';
import {
	VIEW_CONTEXT_DASHBOARD,
	VIEW_CONTEXT_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_MODULE_SETUP,
} from './googlesitekit/constants';
import DashboardEntryPoint from './components/DashboardEntryPoint';

// Initialize the app once the DOM is ready.
domReady( async () => {
	if ( global._googlesitekitLegacyData.admin.resetSession ) {
		clearWebStorage();
	}

	const renderTarget = document.getElementById(
		'js-googlesitekit-dashboard'
	);

	if ( renderTarget ) {
		// Make a separate API request since we cannot use selectors
		// outside the Root component.
		const { authenticated: isAuthenticated } = await API.get(
			'core',
			'user',
			'authentication',
			{},
			{ useCache: false }
		);

		const { setupModuleSlug } = renderTarget.dataset;
		let viewContext = VIEW_CONTEXT_MODULE_SETUP;
		if ( ! setupModuleSlug ) {
			viewContext = isAuthenticated
				? VIEW_CONTEXT_DASHBOARD
				: VIEW_CONTEXT_DASHBOARD_VIEW_ONLY;
		}
		render(
			<Root viewContext={ viewContext }>
				<DashboardEntryPoint setupModuleSlug={ setupModuleSlug } />
			</Root>,
			renderTarget
		);
	}
} );
