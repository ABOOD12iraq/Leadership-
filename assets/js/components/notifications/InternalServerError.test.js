/**
 * InternalServerError component tests.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import InternalServerError from './InternalServerError';
import { render, createTestRegistry } from '../../../../tests/js/test-utils';
import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';

describe( 'InternalServerError', () => {
	const registry = createTestRegistry();

	const internalServerError = {
		id: `module-setup-error`,
		title: 'Internal Server Error',
		description: 'Test error message',
		format: 'small',
		type: 'win-error',
	};

	describe( 'getInternalServerError', () => {
		test( 'should display the notification when the internal server error is set', () => {
			registry.dispatch( CORE_SITE ).setInternalServerError( internalServerError );

			const { container } = render( <InternalServerError />, { registry } );

			expect( container ).toMatchSnapshot();
		} );

		test( 'renders nothing if the internal server error is not set', () => {
			registry.dispatch( CORE_SITE ).setInternalServerError();

			const { container } = render( <InternalServerError />, { registry } );

			expect( container ).toMatchSnapshot();
		} );
	} );
} );
