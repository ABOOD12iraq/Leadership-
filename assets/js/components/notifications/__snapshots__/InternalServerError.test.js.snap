// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalServerError should display the notification when the internal server error is set 1`] = `
<div>
  <section
    class="googlesitekit-publisher-win googlesitekit-publisher-win--small googlesitekit-publisher-win--win-error googlesitekit-publisher-win--is-open"
    id="module-setup-error"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="mdc-layout-grid__cell mdc-layout-grid__cell--span-11-desktop mdc-layout-grid__cell--span-7-tablet mdc-layout-grid__cell--span-3-phone"
        >
          <h3
            class="googlesitekit-heading-2 googlesitekit-publisher-win__title"
          >
            Internal Server Error
          </h3>
          <div
            class="googlesitekit-publisher-win__desc"
          >
            <p>
              <span>
                Test error message
              </span>
            </p>
          </div>
        </div>
        <div
          class=" mdc-layout-grid__cell mdc-layout-grid__cell--span-1 "
        >
          <div
            class="googlesitekit-publisher-win__icons"
          >
            <svg
              aria-labelledby="error-title error-desc"
              fill="#EA4336"
              height="34"
              viewBox="0 0 24 24"
              width="34"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.47 21h15.06c1.54 0 2.5-1.67 1.73-3L13.73 4.99c-.77-1.33-2.69-1.33-3.46 0L2.74 18c-.77 1.33.19 3 1.73 3zM12 14c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1s1 .45 1 1v2c0 .55-.45 1-1 1zm1 4h-2v-2h2v2z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`InternalServerError should render nothing if the internal server error is not set 1`] = `<div />`;
